### DISCUSSION TITLE
Add support for a configureable response when getStaticProps encounters an error

### DISCUSSION THREAD

**Discussion by dcow** [vercel/next.js#38447] [Ideas]
### Describe the feature you'd like to request

Right now, I only have a few options in getStaticProps, essentially: `success`, `redirect`, and `not found`. 

### Describe the solution you'd like

I'd like to have my site return a different status page in different scenarios. For example, a 5XX series error if the CMS can't be reached (instead of `notFound: true`) vs a 404 in the event that I could contact the CMS but the requested information simply does not exist.

### Describe alternatives you've considered

I could use `redirect` to send the user to an appropriate 5XX status page. However, that doesn't seem like the most direct way to achieve the desired result (and I'd have to send any dynamic error message in the query params, I'm guessing). I'm new to nextjs so if this is a completely valid pattern I can implement it. However my inclination would be that a redirect would be used when I want to deliberately send the user to some other resource rather than inform them directly of the status/state/error of the resource they are requesting. If redirecting is the intended flow to use, it's at least odd (without any context) that `notFound` gets a special case.

**Comment by wwwdepot** [comment]
I think it's in the docs, you should throw error when there's server-side problem in getStaticProps.

https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#error-handling-and-revalidation

**Reply by balazsorban44** [reply]
Correct. Next.js already handles errors in these data-fetching scenarios. If the request fails to your CMS, it's not the Next.js application that has failed, so Next.js will keep serving the last correct content, until the data source does not fix the issue, so it should let errors throw.

**Reply by dcow** [reply]
Thanks! While this works specifically for my scenario, for my curiosity and future reference, what would be the intended solution if you *wanted* the user to see an error page rather than the previous content?
